src/moi.jl

```diff
@@ -139,25 +139,93 @@ function process_constraint!(
     return process_constraint!(ir, MOIU.scalarize(f), MOIU.scalarize(s))
 end
 
+struct PresolveResult{T}
+    pd::PresolveData{T}
+end
+
+function get_status(pr::PresolveResult)
+    if pr.pd.status == OPTIMAL
+        return MOI.OPTIMAL
+    elseif pr.pd.status == PRIMAL_INFEASIBLE
+        return MOI.INFEASIBLE
+    elseif pr.pd.status == DUAL_INFEASIBLE
+        return MOI.DUAL_INFEASIBLE
+    else
+        @assert pr.pd.status == NOT_INFERRED
+        return MOI.OPTIMIZE_NOT_CALLED
+    end
+end
+
+function get_optimal_solution(pr::PresolveResult{T}) where {T}
+    if get_status(pr) != MOI.OPTIMAL
+        error(
+            "Presolve did not solve problem to optimality, so you cannot query an optimal solution.",
+        )
+    end
+    @assert pr.pd.solution !== nothing
+    @assert pr.pd.solution.primal_status == FEASIBLE_POINT
+    @assert pr.pd.nrow == pr.pd.ncol == 0
+    orig_sol = _post_crush(pr.pd, pr.pd.solution)
+    @assert orig_sol.primal_status == FEASIBLE_POINT
+    return orig_sol.x
+end
+
+function get_unbounded_ray(pr::PresolveResult{T}) where {T}
+    if get_status(pr) != MOI.DUAL_INFEASIBLE
+        error("Presolve did not prove unboundedness, so you cannot query an unbounded ray.")
+    end
+    @assert pr.pd.solution !== nothing
+    @assert pr.pd.solution.is_primal_ray
+    @assert pr.pd.solution.primal_status == INFEASIBILITY_CERTIFICATE
+    orig_sol = _post_crush(pr.pd, pr.pd.solution)
+    @assert orig_sol.primal_status == INFEASIBILITY_CERTIFICATE
+    return orig_sol.x
+end
+
+function get_infeasibility_certificate(pr::PresolveResult{T}) where {T}
+    if get_status(pr) != MOI.INFEASIBLE
+        error(
+            "Presolve did not prove infeasibility, so you cannot query an infeasibility certificate.",
+        )
+    end
+    error("Not yet implemented. Come back soon!")
+    @assert pr.pd.solution !== nothing
+    @assert pr.pd.solution.is_dual_ray
+    @assert pr.pd.solution.dual_status == INFEASIBILITY_CERTIFICATE
+    return _post_crush(pr.pd, pr.pd.solution)
+end
+
+function post_crush(pr::PresolveResult{T}, x::Vector{T}) where {T}
```

Comment:
`post_crush` should handle points and rays (in case the presolved model is later found to be unbounded/infeasible).

If we stay with just primal presolve, we could have a flag`isray::Bool`.

```suggestion
function post_crush(pr::PresolveResult{T}, x::Vector{T}; isray::Bool=false) where {T}
```